src/main.rs

```diff
@@ -1,173 +1,96 @@
-    if !mapping_path.exists() || refresh_mappings {
-        return true;
-    }
-    false
-}
+mod client;
+mod config;
 
-fn should_refresh_prices(prices_path: &PathBuf, force_prices: bool) -> bool {
-    if !prices_path.exists() || force_prices {
-        return true;
-    }
-    // Check modified time, if > 5 minutes ago, refresh
-    let metadata = prices_path
-        .metadata()
-        .expect("Unable to get metadata for prices.json");
-    let mtime = metadata.modified().expect("Unable to access mtime for prices.json");
-    let duration = SystemTime::now().duration_since(mtime).unwrap();
-    if duration.as_secs() >= 5 * 60 {
-        return true;
-    }
-    false
-}
-
-async fn get_mappings(client: &Client, cache_dir: &PathBuf, refresh_mappings: bool) -> Result<Vec<ItemMapping>, Box<dyn std::error::Error>> {
-    let mapping_path: PathBuf = cache_dir.join("mappings.json");
-
-    if should_refresh_mapping(&mapping_path, refresh_mappings) {
-        // Refresh mapping file if needed
-        let mapping_body = client.get(MAPPING_URL)
-            .send()
-            .await?
-            .text()
-            .await?;
-        fs::create_dir_all(mapping_path.parent().unwrap()).expect("Unable to create directories");
-        fs::write(mapping_path, &mapping_body).expect("Unable to save mapping data");
-        let results = serde_json::from_str(&mapping_body)?;
-        return Ok(results);
-    }
-    let reader = BufReader::new(File::open(mapping_path)?);
-    let results = serde_json::from_reader(reader)?;
-    Ok(results)
-}
-
-async fn get_prices(client: &Client, cache_dir: &PathBuf, force_prices: bool) -> Result<PriceResults, Box<dyn std::error::Error>> {
-    let prices_file = cache_dir.join("prices.json");
-    if should_refresh_prices(&prices_file, force_prices) {
-        // Re-download the prices
-        let prices_body = client.get(PRICES_LATEST_URL)
-            .send()
-            .await?
-            .text()
-            .await?;
-        fs::create_dir_all(prices_file.parent().unwrap())?;
-        fs::write(prices_file, &prices_body).expect("Unable to save prices data");
-        let results: PriceResults = serde_json::from_str(&prices_body)?;
-        return Ok(results);
-    }
-    let reader = BufReader::new(File::open(prices_file)?);
-    let results = serde_json::from_reader(reader)?;
-    Ok(results)
-}
+use anyhow::Result;
+use log::info;
+use num_format::{SystemLocale, ToFormattedString};
 
-fn get_matching_item_ids(name: &String, mappings: &Vec<ItemMapping>) -> HashSet<u32> {
-    let lower = name.to_lowercase();
-    HashSet::from_iter(mappings
-        .iter()
-        .filter(|&mapping| mapping.name.to_lowercase().contains(&lower))
-        .map(|mapping| mapping.id))
+use crate::{
+    client::{Client, ItemMapping, PriceResult},
+    config::Config,
+};
+
+fn get_matching_items<S: AsRef<str>>(
+    name: S,
+    mappings: &[ItemMapping],
+) -> impl Iterator<Item = &ItemMapping> {
```

Comment:
Why does this have the `impl` keyword before the `Iterator`

Reply:
It’s not a concrete return type, it’s saying “the return type will implement the Iterator trait”